rlclientlib/serialization/fb_serializer.h

```diff
@@ -0,0 +1,119 @@
+#pragma once
+#include <vector>
+#include <flatbuffers/flatbuffers.h>
+#include "logger/flatbuffer_allocator.h"
+#include "generated/OutcomeEvent_generated.h"
+#include "generated/RankingEvent_generated.h"
+#include "logger/message_type.h"
+#include "err_constants.h"
+using namespace reinforcement_learning::messages::flatbuff;
+namespace reinforcement_learning { namespace logger {
+  template <typename T>
+  struct fb_event_serializer;
+  template <>
+  struct fb_event_serializer<ranking_event> {
+    using fb_event_t = RankingEvent;
+    using offset_vector_t = typename std::vector<flatbuffers::Offset<fb_event_t>>;
+    using batch_builder_t = RankingEventBatchBuilder;
+
+    static size_t size_estimate(const ranking_event& evt) {
```

Comment:
This size estimation requires for controlling size of queue in memory (before serialization), and there is nothing fb-specific in implementation as well. So, probably it should not be part of serializer<T>? We have static assert that any event have to be inherited from base event class, so probably it is cleaner to add size() method to that interface? In this case we also can get rid of size argument for event_queue::push.